app/controllers/saml_idp_controller.rb

```diff
@@ -70,10 +73,8 @@ def remotelogout
   end
 
   def external_saml_request?
-    return true if request.path.start_with?('/api/saml/authpost')
-
     begin
-      URI(request.referer).host != request.host
+      URI(request.referer).host != request.host || request.referer != complete_saml_url
```

Comment:
I think you're right that it isn't necessary to check that the hosts aren't the same, since we *do* want to log the authpost2022 POST requests as external requests. I'll update!

Reply:
You're right, I only needed to change this when I was trying to use the "final" param, so I'll revert this change.

Reply:
gah I actually think you were right and that this is a bug - requests to `/api/samle/authpostXXXX` _are_ external requests that went through our internal POST process initially. I think instead of excluding the complete URL, why don't we include the initial internal post as a condition as well. What do you think?